sui_core/src/authority.rs

```diff
@@ -643,35 +653,8 @@ impl AuthorityState {
         modules: Vec<CompiledModule>,
     ) -> SuiResult {
         debug_assert!(ctx.digest() == TransactionDigest::genesis());
-        let inputs = Transaction::input_objects_in_compiled_modules(&modules);
-        let input_objects = self.fetch_objects(&inputs, None).await?;
-        // When publishing genesis packages, since the std framework packages all have
-        // non-zero addresses, [`Transaction::input_objects_in_compiled_modules`] will consider
-        // them as dependencies even though they are not. Hence input_objects contain objects
-        // that don't exist on-chain because they are yet to be published.
-        #[cfg(debug_assertions)]
-        {
-            let to_be_published_addresses: HashSet<_> = modules
-                .iter()
-                .map(|module| *module.self_id().address())
-                .collect();
-            assert!(
-                // An object either exists on-chain, or is one of the packages to be published.
-                inputs
-                    .iter()
-                    .zip(input_objects.iter())
-                    .all(|(kind, obj_opt)| obj_opt.is_some()
-                        || to_be_published_addresses.contains(&kind.object_id()))
-            );
-        }
-        let filtered = inputs
-            .into_iter()
-            .zip(input_objects.into_iter())
-            .filter_map(|(input, object_opt)| object_opt.map(|object| (input, object)))
-            .collect::<Vec<_>>();
-
```

Comment:
Why are these deleted?

Reply:
The logic was gone and these asserts were already done inside of the VM, but I added it back for the reasons you stated